test/www/jxcore/test/thalicryptomanagertests.js

```diff
@@ -0,0 +1,61 @@
+require('./mockmobilegetdocpath');
+var test = require('tape');
+var cryptomanager = require('../thali/thalicryptomanager');
+
+
+test('successfully create a new pkcs12 file and return hash value', function (t) {
+  var errorMessage = null,
+      fileLocation = './pkcs12foldergood';
+
+  var hashLength = 16;
+
+  Mobile.storeGetDocumentsPathReturnArguments(errorMessage, fileLocation);
+  
+  cryptomanager.getPublicKeyHash(function (err, publicKeyHash) {
+    t.equal(err, null);
+    t.equal(publicKeyHash.length, hashLength);
+    t.end();
+  });
+});
+
+test('successfully read a previous pkcs12 file and return hash value', function (t) {
+  var errorMessage = null,
```

Comment:
It's not safe to rely on test execution order. If the tests get moved around or the test running decides to run the tests in a different order (happens all the time) then this test will look like it passed but in fact it didn't do what you want. What you really need to do is first call the PKCS12 generation function yourself directly. Then read in the file and generate the hash. Then call getPublicKeyHash and makes sure it returns the same hash so you know it read the same document.